### ISSUE TITLE
Problem in handleFlags,adp-method

### ISSUE THREAD

**Issue by j-harbin** [dankelley/oce#1914]
As discussed in Zoom with DK

![Screen Shot 2022-01-11 at 10 08 23 AM](https://user-images.githubusercontent.com/58750538/148957919-6bd17d82-789f-4fd4-b8d9-89d4a04d5f39.png)

**Comment by dankelley** [comment]
I'll fix this sometime this afternoon.  I have no idea why I'm looping through `j` in this code.  I'll fix that up, and do some tests.  And I'll also clean up the debugging output, to give *something* for some functions that are now silent, and to give *less* for those that are just too wordy ... um... for words.

**Comment by dankelley** [comment]
I think it works now. Please check, @j-harbin, and close if so (or comment if not).  Below is the info from my commit

commit 2fb481a220c8db7a3a26b009e437e1409304bb09
Author: dankelley <kelley.dan@gmail.com>
Date:   Tue Jan 11 15:01:22 2022 -0400

    See https://github.com/dankelley/oce/issues/1914
    
    1. The is.raw() check is correct now, so issue 1914 has been addressed.
    2. I made the debugging output be more informative, yet less verbose.
    3. The diagram made by example('handleFlags,adp-method') now has an arrow
       pointing out a spot that changes (subtly) when flags are applied.
    4. There is now a test-suite component on this, as below (from
       tests/testthat/test_flags.R)
    
    ```R
    test_that("adp handleFlag gives error for raw data (issue 1914)", {
        data(adp)
        v <- adp[["v"]]
        i2 <- array(FALSE, dim=dim(v))
        g <- adp[["g", "numeric"]]
        # Thresholds on percent "goodness" and error "velocity"
        G <- 25
        V4 <- 0.45
        for (k in 1:3)
            i2[,,k] <- ((g[,,k]+g[,,4]) < G) | (v[,,4] > V4)
        # Can apply flags to velocity, because it is numeric
        a <- initializeFlags(adp, "v", 2)
        b <- setFlags(a, "v", i2, 3)
        expect_silent(c <- handleFlags(b, flags=list(3), actions=list("NA")))
        # Cannot apply flags to amplitude, because it is raw
        a <- initializeFlags(adp, "a", 2)
        b <- setFlags(a, "a", i2, 3)
        expect_error(c <- handleFlags(b, flags=list(3), actions=list("NA")),
            "use adpConvertRawToNumeric")
    })
    ```

**Comment by dankelley** [comment]
Dear reporter,

Do you think this issue (as defined by its title) has been addressed? If so, please close it. If not, please add a comment explaining what remains to be done. *Thanks!*

PS.  This is a standardized reply.

**Comment by dankelley** [comment]
This issue has been closed, since it seems to have been addressed, and since  discussion has stalled. The original reporter should feel free to reopen it, though, or to open new issues that might be related. Thanks.

PS.  This is a standardized reply.
